pkg/api/v1/router_server_components_test.go

```diff
@@ -517,17 +517,41 @@ func TestIntegrationServerUpdateComponents(t *testing.T) {
 		return err
 	})
 
-	componentSliceFixture := func() []serverservice.ServerComponent {
-		s := serverservice.ServerComponentSlice{}
+	// fixtures given to test cases below
+	var componentFixture []serverservice.ServerComponent
 
-		for _, c := range servers[0].Components {
+	var serverFixture serverservice.Server
+
+	// The component fixture targeted in test cases below
+	fixtureComponentName := "My Lucky Fin"
+	fixtureComponentVendor := "Barracuda"
+	fixtureComponentSerial := "Right"
+
+	// identify component and server fixture for test
+	for _, server := range servers {
+		server := server
+		for _, c := range server.Components {
```

Comment:
same question here with the component. In this case, it's better to iterate the id's... e.g.:

```
for idx := range server.Components {
```

and then copy set the memory reference in a variable.

Reply:
I think the `c := c` was from a previous change in the code where the object was being stored as a pointer in a slice, where I was being cautious about the common range temporary variable gotcha, that along with trying not to type as much to reference each of the struct fields - `c.Name == fixtureComponentName`. 

Thanks for pointing it out, I've pushed a fix https://github.com/metal-toolbox/hollow-serverservice/pull/118/commits/81e7ea5e64883b10fc46374399ed56831c94c70f to remove the range variable copying which isn't necessary since the component objects are being stored by value.